packages/extension/src/page/PostMessageProvider.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 @polkadot/extension authors & contributors
```

Comment:
I have just been scrolling through, so no in-depth comments yet - however, it would be good if we can actually move the provider (as to be injected) to `extension-provider` so it is actually easily re-usable by others as well.

(I can see that this file, as-is, can be shared - just the endpoint can be custom for a "quick start")

Reply:
And yes... need to do _exactly_ the same with the injected signer. (In thinking about these 2, it may end up being in `extension-inject` then as sub-folders.) So extensions do the following -

- include the package
- use the top-level interface to inject into the page
- create an instance of the signer for signing (or do their own)
- create an instance of the provider (or ignore, or do their won)

One self-contained bundle.

Reply:
> it would be good if we can actually move the provider (as to be injected) to extension-provider so it is actually easily re-usable by others as well.

PostMessageProvider only sends messages to the window, it heavily relies on the logic (and message format) of page.js/content.js so that the messages are sent to the extension and received from the extension; as such I don't think it's very reusable by itself. Same for the injected signer, which basically only does `sendRequest('extrinsic.sign', payload);`. What do you think?

Reply:
I would suggest looking at how other extensions are using this - as it is, it is being used as-is. You do have a good suggestion buried in there though to make the message names slightly less generic for these case, i.e so it is explicit where it comes from and what it is for.